### ISSUE TITLE
Runtime error on macOS pyInstaller application - NotADirectoryError for base_library.zip

### ISSUE THREAD

**Issue by jeffro84** [pyinstaller/pyinstaller#4235]
macOS Mojave 10.14.4
Python 3.7.3
PyInstaller 3.4 (also happens on latest dev version)

I have a working quart application, but it fails at runtime after being packaged with pyinstaller.  I have created a minimal version of the application and it still causes the error.

minimal.py
--------
```
import quart
app = quart.Quart(__name__)

if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8080, debug=False)
```

minimal.spec
--------
```
# -*- mode: python -*-
  
block_cipher = None

options = [ ('v', None, 'OPTION') ]
a = Analysis(['../../minimal.py'],
             pathex=['.'],
             binaries=[],
             datas=[],
             hiddenimports=['quart'],
             hookspath=[],
             runtime_hooks=[],
             excludes=[],
             cipher=block_cipher)
pyz = PYZ(a.pure, a.zipped_data,
             cipher=block_cipher)
exe = EXE(pyz,
          a.scripts,
          a.binaries,
          a.zipfiles,
          a.datas,
          name='Minimal',
          debug=False,
          strip=False,
          upx=True,
          console=True )
```

See attached output files.  PyInstaller runs successfully [out.txt](https://github.com/pyinstaller/pyinstaller/files/3192784/out.txt).  But when I try to run the application, it fails with a NotADirectoryError [error.txt](https://github.com/pyinstaller/pyinstaller/files/3192783/error.txt)

I determined that it is complaining about base_library.zip, which of course is not a directory.  But it tries to access it as if it were:
`NotADirectoryError: [Errno 20] Not a directory: '/var/folders/gl/40zhbkc142v7bkcpfn7jq0_h0000gn/T/_MEI5VmDiU/base_library.zip/encodings/__init__.pyc'`

So that is why I concluded there might be a bug here -- either pyinstaller forgot to extract base_library.zip or it forgot to update the path after doing so.  I tried changing the runtime_tmpdir but it gives the same error (with a different leading path).

Any thoughts or suggestions?

**Comment by htgoebel** [comment]
You would have helped me, if you'd appended the few lines of error-lug quite here. Thank for wasting my time.
```
$ ./dist/Minimal
Running on http://0.0.0.0:8080 (CTRL + C to quit)
Traceback (most recent call last):
  File "minimal.py", line 7, in <module>
  File "site-packages/quart/app.py", line 1393, in run
  File "asyncio/runners.py", line 43, in run
  File "asyncio/base_events.py", line 584, in run_until_complete
  File "site-packages/hypercorn/asyncio/__init__.py", line 30, in serve
  File "site-packages/hypercorn/asyncio/run.py", line 174, in worker_serve
  File "site-packages/hypercorn/utils.py", line 90, in observe_changes
  File "pathlib.py", line 1161, in stat
NotADirectoryError: [Errno 20] Not a directory: '/var/folders/gl/40zhbkc142v7bkcpfn7jq0_h0000gn/T/_MEI5VmDiU/base_library.zip/encodings/__init__.pyc'
[46853] Failed to execute script minimal
```

Obviously  hypercorn/utils.py", line 90 is not adopted to be used when frozen. Please read the PyInstaller manual and get in tough with the quthors of hypercorn.

**Comment by jeffro84** [comment]
Hi htgoebel, thanks for finding the time to look at this.

If you look at my original post, you see I did indeed attach the full error.txt output which mirrors what you pasted in your reply.  I followed all of the instructions before posting anything here precisely because I didn't want to waste anyone's time.  And yes, before even considering posting here, I also spent much time going through the pyinstaller manual, to no avail.

Unfortunately, unlike you, I do not have intimate knowledge of the pyinstaller codebase.  I'm glad it was an obvious issue for you so it didn't waste too much of your time.

Peace
